devices/YarpRobotLoggerDevice/include/BipedalLocomotion/YarpRobotLoggerDevice.h

```diff
@@ -166,6 +173,11 @@ class YarpRobotLoggerDevice : public yarp::dev::DeviceDriver,
     std::vector<std::string> m_textLoggingSubnames;
     std::vector<std::string> m_codeStatusCmdPrefixes;
 
+    // for realtime data streaming
+    bool streamRealTime{true};
+    yarp::os::BufferedPort<yarp::os::Bottle> realtimeLoggingPort;
+    BipedalLocomotion::YarpUtilities::VectorsCollection* externalSignalCollection;
```

Comment:
Please, avoid storing this as a class parameter. Since it is the output of ``readData``, the pointed comment lifetime cannot match the lifetime of the class. It is safer to have this as a local variable only.

Reply:
The reason why I originally made this a class parameter has to do with the need for the data twice. Once for the regular logging to the .mat file and the other read for sending to the robot-log-visualizer, I noticed that if I all signal.client.readData(false) twice, the second time would not receive the data. Hence, I made it a class parameter and called signal.client.readData(false) once and read from the variable twice.

Reply:
Fixed in [c3c79ad](https://github.com/ami-iit/bipedal-locomotion-framework/commit/c3c79adf8f8b028d5aa7a0e6125282466447e4b0)

Reply:
After face-to-face discussion with @S-Dafarra, additional work was done in commit ([c3c79ad](https://github.com/ami-iit/bipedal-locomotion-framework/commit/c3c79adf8f8b028d5aa7a0e6125282466447e4b0) to make vector collection server and the buffer manager in the same function. Additionally, hard-coded values related to the naming of the variables were replaced in a new header file